akka-persistence-tck/src/main/scala/akka/persistence/query/EventsByTagQuerySpec.scala

```diff
@@ -0,0 +1,196 @@
+package akka.persistence.query
+
+import scala.concurrent.duration._
+
+trait EventsByTagQuerySpec { _: QuerySpec ⇒
+
+  "EventsByTagQuery" must {
+    "not produce anything given an empty tag" in {
+      withEventsByTag()(tag = "", offset = 0L) { tp ⇒
+        tp.request(10)
+        tp.expectNoMsg(100.millis)
+        tp.cancel()
+      }
+    }
+
+    "not produce anything if there aren't any events matching the tag" in {
+      withEventsByTag()(tag = "unknown-tag", offset = 0L) { tp ⇒
+        tp.request(10)
+        tp.expectNoMsg(100.millis)
+        tp.cancel()
+      }
+    }
+
+    "find current events" in {
+      persist(1, 2, pid, tag)
+
+      withEventsByTag()(tag, offset = 0L) { tp ⇒
+        tp.request(2)
+        tp.expectNext(EventEnvelope(1L, pid, 1L, "a-1"))
+        tp.expectNext(EventEnvelope(2L, pid, 2L, "a-2"))
+        tp.expectNoMsg(100.millis)
+        tp.cancel()
+      }
+    }
+
+    "find new events" in {
+      val pidA = pid
+      val pidB = nextPid
+
+      persist(1, 1, pidA, tag)
+
+      withEventsByTag()(tag, offset = 0L) { tp ⇒
+        tp.request(2)
+        tp.expectNext(EventEnvelope(1L, pidA, 1L, "a-1"))
+        tp.expectNoMsg(100.millis)
+
+        persist(1, 2, pidB, tag)
+
+        tp.expectNext(EventEnvelope(2L, pidB, 1L, "a-1"))
+        tp.expectNoMsg(100.millis)
+        tp.request(10)
+        tp.expectNext(EventEnvelope(3L, pidB, 2L, "a-2"))
+        tp.expectNoMsg(100.millis)
+        tp.cancel()
+      }
+    }
+
+    "find events from offset" in {
+      val pidA = pid
+      val pidB = nextPid
+      val pidC = nextPid
+
+      persist(1, 1, pidA)
+      persist(2, 2, pidA, tag)
+      persist(3, 3, pidA, tag)
+      persist(1, 1, pidB)
+      persist(2, 2, pidB, tag)
+      persist(1, 1, pidC, tag)
+
+      withEventsByTag()(tag, offset = 2L) { tp ⇒
+        tp.request(10)
+        tp.expectNext(EventEnvelope(2L, pidA, 3L, "a-3"))
+        tp.expectNext(EventEnvelope(3L, pidB, 2L, "a-2"))
+        tp.expectNext(EventEnvelope(4L, pidC, 1L, "a-1"))
```

Comment:
what assumptions about the offset does this test make? the offset doesn't have to be a sequence number, it's up to the plugin to define what the `Long` represents.